.github/workflows/benchmark_execution.yml

```diff
@@ -95,13 +102,15 @@ jobs:
       - runner-group=${{ inputs.runner-group }}
       - environment=${{ inputs.runner-env }}
       - machine-type=${{ matrix.benchmark.device_name }}
-    outputs:
-      benchmark-results-gcs-artifact-dir: ${{ steps.upload.outputs.benchmark-results-gcs-artifact-dir }}
     env:
       DEVICE_NAME: ${{ matrix.benchmark.device_name }}
       PLATFORM_ARCH: ${{ matrix.benchmark.host_environment.platform }}-${{ matrix.benchmark.host_environment.architecture }}
       E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR: ${{ inputs.e2e-test-artifacts-gcs-artifact-dir }}
       E2E_TEST_ARTIFACTS_DIR: ${{ inputs.e2e-test-artifacts-dir }}
+      BENCHMARK_RESULTS_DIR: benchmark-results
+    outputs:
+      benchmark-results-dir: ${{ env.BENCHMARK_RESULTS_DIR }}
+      benchmark-results-gcs-artifact-dir: ${{ steps.upload.outputs.benchmark-results-gcs-artifact-dir }}
```

Comment:
I thought there were problems specifying outputs from matrix jobs. Does this work because this matrix is only actually one job? Can you add a comment to that effect by the matrix specification above? It says it's dynamically generated, but not that it only results in one job (which is a bit of a confusing hack around GitHub Actions' lack of proper dynamic jobs)

Reply:
Added comments above. Yes, the behavior is as you said.